src/helpers/Blocks/Blocks.js

```diff
@@ -152,44 +152,32 @@ export function addBlock(formData, type, index) {
 }
 
 export function mutateBlock(formData, id, value) {
-  const { settings } = config;
+  return addBlockBefore(formData, id, value)[1];
+}
+
+export function addBlockBefore(formData, id, value) {
   const blocksFieldname = getBlocksFieldname(formData);
   const blocksLayoutFieldname = getBlocksLayoutFieldname(formData);
-  const index = formData[blocksLayoutFieldname].items.indexOf(id) + 1;
-
-  // Test if block at index is already a placeholder (trailing) block
-  const trailId = formData[blocksLayoutFieldname].items[index];
-  if (trailId) {
-    const block = formData[blocksFieldname][trailId];
-    if (!blockHasValue(block)) {
-      return {
-        ...formData,
-        [blocksFieldname]: {
-          ...formData[blocksFieldname],
-          [id]: value || null,
-        },
-      };
-    }
-  }
+  const index = formData[blocksLayoutFieldname].items.indexOf(id);
 
-  const idTrailingBlock = uuid();
-  return {
-    ...formData,
-    [blocksFieldname]: {
-      ...formData[blocksFieldname],
-      [id]: value || null,
-      [idTrailingBlock]: {
-        '@type': settings.defaultBlockType,
+  const newBlockId = uuid();
+  return [
+    newBlockId,
+    {
+      ...formData,
+      [blocksFieldname]: {
+        ...formData[blocksFieldname],
+        [newBlockId]: value || null,
+      },
+      [blocksLayoutFieldname]: {
+        items: [
+          ...formData[blocksLayoutFieldname].items.slice(0, index),
+          newBlockId,
```

Comment:
Replacing the current block with newer block created. I like this :)

Reply:
:wink: